File: csharp/main/LiteralTimestamp.cs
using System;

namespace Deltix.Luminary
{
	public sealed class LiteralTimestamp : Literal, ILiteral<String>
	{
		public const String MinValueAsString = "#MIN";
		public const String MaxValueAsString = "#MAX";

		public LiteralTimestamp(String value)
			: base(LiteralKind.Timestamp)
		{
			Value = value;
		}

		public String Value { get; }

		public override String ToString()
		{
			return $"\"{Value}\"";
		}
	}
}

File: java/src/main/java/deltix/luminary/LiteralTimestamp.java
package deltix.luminary;

public class LiteralTimestamp extends Literal {
    private String value;

    public LiteralTimestamp(String value) {
        super(LiteralKind.TIMESTAMP);
        this.value = value;
    }

    public String getValue() {
        return value;
    }

    @Override
    public String toString() {
        return "\"" + value + "\"";
    }

    public static final String MIN_VALUE_AS_STRING = "#MIN";
    public static final String MAX_VALUE_AS_STRING = "#MAX";
}
